lib/spack/spack/spec.py

```diff
@@ -887,6 +989,13 @@ def concretized(self):
 
 
     def flat_dependencies(self, **kwargs):
+        flat_deps = DependencyMap()
+        for name, depspec in self.flat_dependencies_with_deptype(**kwargs).items():
+            flat_deps[name] = depspec.spec
+        return flat_deps
+
+
+    def flat_dependencies_with_deptype(self, **kwargs):
```

Comment:
I'm a little worried about all the `_with_deptype` suffixed things, but I don't have an obviously better idea off the top of my head.